PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx

```diff
@@ -3509,153 +3509,33 @@ Bool_t AliCaloPhotonCuts::CheckDistanceToBadChannel(AliVCluster* cluster, AliVEv
-      if ((coldiff + rowdiff <= fMinDistanceToBadChannelSwapping )) return kTRUE;
-
-      //cout << "not within distanceToBadChannel!" << endl;
-    }
+  } else if( fClusterType == 2){ // PHOS case
+    Int_t relid[4];
+    fGeomPHOS->AbsToRelNumbering(CellID,relid);
+    if(relid[1]!=0) AliFatal("PHOS CPV in PHOS cell array (rotation background method)?");
+    if(relid[0]>=fGeomPHOS->GetNModules() || relid[0]<0 || !fPHOSBadChannelsMap[relid[0]]) return kTRUE;
+    iBadCell = (Int_t) ((TH2I*)fPHOSBadChannelsMap[relid[0]])->GetBinContent(relid[2],relid[3]);
   }
 
-  //check bad cells in neighboring SM only if within chosen distanceToBadChannel from maxEnergyCell the next SM could be reached
-  if(checkNextSM) {
-    // In case of a shared cluster, index of SM in C side, columns start at 48 and ends at 48*2-1
-    // C Side impair SM, nSupMod%2=1;A side pair SM nSupMod%2=0
-    if( fClusterType == 1 || fClusterType == 4){
-      if(CelliMod%2){
-        nMinCols = largestCellicol - distanceForLoop + AliEMCALGeoParams::fgkEMCALCols;
-        nMaxCols = AliEMCALGeoParams::fgkEMCALCols;
-
-        CelliMod -= 1;
-        largestCellicol += AliEMCALGeoParams::fgkEMCALCols;
-      }else{
-        nMinCols = 0;
-        nMaxCols = largestCellicol + distanceForLoop - AliEMCALGeoParams::fgkEMCALCols;
-
-        CelliMod += 1;
-        largestCellicol -= AliEMCALGeoParams::fgkEMCALCols;
-      }
-    }else if( fClusterType == 2 ){
-     // nMaxRows = 64;
-     // nMaxCols = 56;
-    }
-    //cout << "largestCell: " << largestCellID << ",mod: " << CelliMod << ",col: " << largestCellicol << ",row: " << largestCellirow << endl;
-    //cout << "distanceForLoop: " << distanceForLoop << ",nMinRows: " << nMinRows << ",nMaxRows: " << nMaxRows << ",nMinCols: " << nMinCols << ",nMaxCols: " << nMaxCols << endl;
-    for (Int_t irow = nMinRows;irow < nMaxRows;irow++)
-    {
-      for (Int_t icol = nMinCols;icol < nMaxCols;icol++)
-      {
-        Int_t iBadCell = 0;
-        if( (fClusterType == 1 || fClusterType == 4) && CelliMod<fEMCALBadChannelsMap->GetEntries()){
-          iBadCell = (Int_t) ((TH2I*)fEMCALBadChannelsMap->At(CelliMod))->GetBinContent(icol,irow);
-        }else if( fClusterType == 2 && fPHOSBadChannelsMap[CelliMod+1]){
-          iBadCell = (Int_t) ((TH2I*)fPHOSBadChannelsMap[CelliMod+1])->GetBinContent(icol,irow);
-        }
-        //cout << "CelliMod: " << CelliMod << ",iBadCell: " << iBadCell << ",icol: " << icol << ",irow: " << irow << endl;
-        if(iBadCell==0) continue;
-
-        rowdiff = TMath::Abs( largestCellirow - irow ) ;
-        coldiff = TMath::Abs( largestCellicol - icol ) ;
-        //cout << "rowdiff: " << rowdiff << ",coldiff: " << coldiff << endl;
-        if ((coldiff + rowdiff <= fMinDistanceToBadChannelSwapping )) return kTRUE;
-        //cout << "not within distanceToBadChannel!" << endl;
-      }
-    }
-  }
 
+  if(iBadCell != 0) return kTRUE;
```

Comment:
Shouldn't we have iBadCell > 0 here to protect against the initialization of iBadCell = -1?

Reply:
returning true actually means that the cell is bad (I implemented that similar to the existing DistanceToBadChannel function). So this means that if the cell is good iBadCell == 0. And otherwise its bad (so iBadCell !=0 covers the cases where the cell is bad including the initialization of -1).